### DISCUSSION TITLE
Why can't I add data to the tables created in this example from the documentation?

### DISCUSSION THREAD

**Discussion by jbhanks** [sqlalchemy/sqlalchemy#10265] [Usage Questions] [answered]
I am trying to follow this section of the documentation: https://docs.sqlalchemy.org/en/20/orm/inheritance.html#relationships-with-joined-inheritance

If I follow it as shown, I get this error:when I try to insert data:

```
    Company(name="EvilCorp")
TypeError: __init__() missing 2 required positional arguments: 'id' and 'employees'
```

I was able to solve the issue by adding `init=False` to the required columns, but shouldn't that be in the example?

**Comment by CaselIT** [comment]
Hi,

I can't reproduce. What sqlalchemy version are you using?

**Reply by jbhanks** [reply]
Sorry, my mistake. It was actually working but I got confused. I did need to add `init=False`, which is not in the example though.
Unfortunately after I did that I accidentally deleted something that caused another problem (hence all my edits)

**Reply by CaselIT** [reply, marked as answer]
The example does not use MappedAsDataclass, so there is no need of init=False.

If want to use dataclasses, you will have the follow standard dataclasses rules

**Reply by jbhanks** [reply]
Should [reading this](https://docs.sqlalchemy.org/en/20/orm/dataclasses.html) have been sufficient to know that? I see that all the examples there use it for the PK column, but I'm not sure I would have understood it was essential without a deep read. Some sort of bullet-point list of differences between the different ways of making a model would be really helpful for people like me who are trying to learn all this at once.

**Reply by CaselIT** [reply]
I think the assumption in the docs is that since dataclasses are a std-lib feature people would be familiar on how they work. So if a field is not wanted in the init one would use the same args as when defining a normal dataclass
